pulsar-broker/src/main/java/org/apache/pulsar/broker/namespace/NamespaceService.java

```diff
@@ -825,6 +827,20 @@ public ServiceUnitId getServiceUnitId(TopicName topicName) throws Exception {
             // NoNode means there are no persistent topics for this namespace
         }
 
+        // Non-persistent topics don't have managed ledges so we have to retrieve them from local cache.
+		synchronized (pulsar.getBrokerService().getMultiLayerTopicMap()) {
```

Comment:
This will only answer with the non-persistent topics served by the broker that it's handling the particular request. However, topics for a single namespace can be spread across multiple brokers. 

We would need to do something like this ( https://github.com/apache/incubator-pulsar/blob/master/pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/NonPersistentTopics.java#L170 ) to get the correct list of non-persistent topics.

I think the best option is to actually use the same API to get the list of non-persistent topics through HTTP. `GET /admin/v2/non-persistent/{tenant}/{namespace}`